server/@types/ui/index.d.ts

```diff
@@ -0,0 +1,130 @@
+import { RoshRisks, RiskTier, FlagsEnvelope, Mappa } from '@approved-premises-shared'
+
+// A utility type that allows us to define an object with a date attribute split into
+// date, month, year (and optionally, time) attributes. Designed for use with the GOV.UK
+// date input
+export type ObjectWithDateParts<K extends string | number> = { [P in `${K}-${'year' | 'month' | 'day'}`]: string } & {
+  [P in `${K}-time`]?: string
+} & {
+  [P in K]?: string
+}
+
+export type BookingStatus = 'arrived' | 'awaiting-arrival' | 'not-arrived' | 'departed' | 'cancelled'
+
+export type TaskNames = 'basic-information' | 'type-of-ap'
+
+export interface HtmlAttributes {
+  [key: string]: string
+}
+
+export interface TextItem {
+  text: string
+}
+
+export interface HtmlItem {
+  html: string
+}
+
+export type TableCell = { text: string; attributes?: HtmlAttributes } | { html: string }
+export interface TableRow {
+  [index: number]: TableCell
+}
+
+export interface RadioItem {
+  text: string
+  value: string
+  checked?: boolean
+}
+
+export interface CheckBoxItem {
+  text: string
+  value: string
+  checked?: boolean
+}
+
+export interface SelectOption {
```

Comment:
Again, not for now, but it would be ace if we could break some of this out into a types library for GOV.UK frontend that we could publish via `DefinitelyTyped`. It's wild that this doesn't exist already

Reply:
Yeah I was thinking this, it would be easy now we're using paths in the tsconfig

Reply:
That would be nice!